dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -169,8 +163,14 @@ def __init__(self,*, model_identifier, restart=False, savestate=10):
         self.model_graph = CONFIG.model_graph
         self.step_count = CONFIG.step_count
         self.step_target = CONFIG.step_target
+        self.checkpoint_period = CONFIG.checkpoint_period
+        self.milestones = CONFIG.milestones
         self.steering_parameters = CONFIG.steering_parameters
 
+        # Code version.
+        self.version = Path('version.md').read_text().splitlines()[0]
```

Comment:
I like the idea of saving the software version along with the model output. but how practical is it to use git commit hash? It is not clear how this works in practice. For example, who runs the `regen_version.sh` file to store the commit hash, and how often is it run? Do the users use git to run experiments? Instead, we could use the software version specified in `pyproject.toml` as `version = "0.1.0"`.

Reply:
Based on our discussion, I will add some guidance for this in the `CONTRIBUTING.md` file.

In short, the version is stored in a tracked file. This makes it independent of the repository: the version file can be shipped with the built code (outside the repo) and the code would still work. However, tracking the file means it cannot use the 'current commit hash', because committing this version file (with the current hash) will create a newer hash.
Instead, we should run the `regen_version.md` script before any release (or other 'important code version'). For example as part of the latest commit before tagging for release. It may be possible to do this using git actions, or we may have to do this manually, in which case it should be very clear from the `CONTRIBUTING.md`

I understand this may not be the ideal approach, but I could not find a better approach :(

Reply:
I also found that the relative path of Path('version.md') causes problems. Can you please make it absolute?

Reply:
I changed code version to process version to reflect that this captures other aspects of the process as well.

Tbh, I don't really know how best to get the path of the package root (as opposed to the model save directory). Will discuss.

Reply:
> Tbh, I don't really know how best to get the path of the package root (as opposed to the model save directory). Will discuss.

`parent_path = Path(__file__).resolve().parents`, `parents` is a list, see [doc](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents)